### ISSUE TITLE
(yuefengz): the updates to var and weighted can not be batched 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#3794]
together if we fetch their updated values here. Consider calculating
new values and delaying the updates.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [693b339ab2f062ec5bbb29f976c5d1fd94fbffa5](https://github.com/imdone/tensorflow/commit/693b339ab2f062ec5bbb29f976c5d1fd94fbffa5) by Francois Chollet fchollet@google.com*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/python/keras/_impl/keras/layers/normalization.py#L472-L479
